dialect.go

```diff
@@ -1,364 +1,41 @@
 package goose
 
 import (
-	"database/sql"
 	"fmt"
+
+	"github.com/pressly/goose/v3/internal/dialect"
 )
 
-// SQLDialect abstracts the details of specific SQL dialects
-// for goose's few SQL specific statements
-type SQLDialect interface {
```

Comment:
To evolve `goose` I think we need a more sane way to decouple the dialect queries from the package itself.

We also mix raw queries and `db *sql.DB` queries which leave it up to the caller to deal with `*sql.Rows` directly, that's not great.